spec/lib/healthchecks/queue_dead_set_check_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'sidekiq/testing'
+require 'okcomputer'
+require 'healthchecks'
+require 'scholarsphere/redis_config'
+
+RSpec.describe HealthChecks::QueueDeadSetCheck do
+  before do
+    Sidekiq::Testing.disable!
+    redis_config = Scholarsphere::RedisConfig.new
+    Sidekiq.configure_server do |config|
+      config.redis = redis_config.to_hash
+    end
+
+    Sidekiq.configure_client do |config|
+      config.redis = redis_config.to_hash
+    end
+
+    ds = Sidekiq::DeadSet.new
+    ds.clear
+  end
+
+  describe '#check' do
+    context 'when there are no messages in deadset' do
+      it 'returns no failure' do
+        hc = described_class.new
+        hc.check
+        expect(hc.failure_occurred).to be nil
+      end
+    end
+
+    context 'when there are messages in deadset' do
+      before do
+        serialized_job = Sidekiq.dump_json(jid: '123123', class: 'SomeWorker', args: [])
```

Comment:
@whereismyjetpack I wonder if a "nicer" way of doing this would be:
``` ruby 
class DeadSetTestJob < ApplicationJob
  queue_as :test_queue

  def perform
    raise StandardError, "I'm failing on purpose!"
  end
end
```
Would that go in the failure queue?

Reply:
That would go in the failure queue, but in sidekiq "failure" is just for statistics, we'd have to wait for retries before it got into the deadset -- which is why i decided to short circuit that and plop something on deadset.